web-common/src/features/entity-management/file-artifacts-store.ts

```diff
@@ -1,131 +1,241 @@
+        case ResourceKind.Chart:
+        case ResourceKind.Dashboard:
+          this.setResource(resource);
+          break;
+      }
     }
-    callback(state.entities[path]);
-    return state;
-  });
-};
-
-const fileArtifactsEntitiesReducers = {
-  setName(path: string, name: string) {
-    createOrUpdateFileArtifact(path, (entityData) => {
-      entityData.name = name;
+  }
+
+  public updateFile(filePath: string) {
+    if (get(this.files)[filePath]) return;
+    this.artifacts[filePath] ??= new FileArtifact(filePath);
+    this.files.update((f) => {
+      f[filePath] = false;
+      return f;
     });
-  },
+  }
+
+  public deleteFile(filePath: string) {
+    delete this.artifacts[filePath];
+    this.files.update((f) => {
+      delete f[filePath];
+      return f;
+    });
+  }
+
+  public setResource(resource: V1Resource) {
+    this.updateArtifact(resource);
+    this.updateReconciling(resource);
+  }
 
-  setErrors(affectedPaths: Array<string>, errors: Array<V1ReconcileError>) {
-    const errorsForPaths = new Map<string, Array<V1ReconcileError>>();
-    affectedPaths.forEach((affectedPath) =>
-      errorsForPaths.set(correctFilePath(affectedPath), []),
+  public updateReconciling(resource: V1Resource) {
+    this.files.update((f) => {
+      resource.meta?.filePaths?.map(correctFilePath).forEach((filePath) => {
+        f[filePath] =
+          resource.meta?.reconcileStatus ===
+          V1ReconcileStatus.RECONCILE_STATUS_RUNNING;
+      });
+      return f;
+    });
+  }
+
+  public updateArtifact(resource: V1Resource) {
+    resource.meta?.filePaths?.map(correctFilePath).forEach((filePath) => {
+      this.artifacts[filePath] ??= new FileArtifact(filePath);
+      this.artifacts[filePath].updateResource(resource);
+    });
+  }
+
+  public deleteResource(resource: V1Resource) {
+    resource.meta?.filePaths?.map(correctFilePath).forEach((filePath) => {
+      this.artifacts[filePath]?.deleteResource();
+    });
+  }
+
+  // Selectors
+
+  public getFileArtifact(filePath: string) {
+    return getSubStore(
+      this.files,
+      this.artifacts,
+      filePath,
+      // Dummy store to not break component code
+      new FileArtifact(filePath),
     );
+  }
 
-    errors.forEach((error) => {
-      const filePath = correctFilePath(error.filePath);
+  public getLastStateUpdatedOn(filePath: string) {
+    return this.artifacts[correctFilePath(filePath)]?.lastStateUpdatedOn;
+  }
 
-      // empty models should not show error
-      if (
-        filePath.endsWith(".sql") &&
-        error.message.endsWith("No statement to prepare!")
-      ) {
-        return;
-      }
+  public getResourceNameForFile(
+    filePath: string,
+  ): Readable<V1ResourceName | undefined> {
+    return derived(this.getFileArtifact(filePath), (artifact, set) =>
+      artifact.name.subscribe(set),
+    );
+  }
 
-      if (!errorsForPaths.has(filePath)) {
-        errorsForPaths.set(filePath, []);
+  public getReconcilingItems() {
```

Comment:
Naming suggestion: `getReconcilingFiles`

Reply:
It doesnt actually returns files, rather resource names. `getReconcilingResourceNames` makes more sense.